backend/services/media_generation_service.py

```
@@ -0,0 +1,421 @@
+            
+        except Exception as e:
+            logger.error(f"❌ Video placeholder generation failed: {e}")
+            raise
+    
+    async def _generate_image_fallback(self, params: Dict, filename: str) -> Dict:
+        """Fallback image generation without PIL"""
+        try:
+            # Create simple HTML canvas-based image (base64 encoded)
+            # This is a basic fallback - in production, use PIL or external service
+            
+            canvas_html = f"""
```

Ruff (0.12.2)

Local variable `canvas_html` is assigned to but never used

Remove assignment to unused variable `canvas_html`

(F841)